src/test/java/org/opensearch/neuralsearch/plugin/query/NeuralQueryIT.java

```diff
@@ -0,0 +1,269 @@
+    private static final String TEST_KNN_VECTOR_FIELD_NAME_1 = "test-knn-vector-1";
+    private static final String TEST_KNN_VECTOR_FIELD_NAME_2 = "test-knn-vector-2";
+    private static final String TEST_KNN_VECTOR_FIELD_NAME_NESTED = "nested.knn.field";
+
+    private static final int TEST_DIMENSION = 384;
+    private static final SpaceType TEST_SPACE_TYPE = SpaceType.L2;
+    private final float[] testVector = createRandomVector(TEST_DIMENSION);
+    private final AtomicReference<String> modelId = new AtomicReference<>();
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        modelId.compareAndSet(null, prepareModel());
+    }
+
+    private void maybeInitializeIndex(String indexName) throws IOException {
+        if (TEST_BASIC_INDEX_NAME.equals(indexName) && !indexExists(TEST_BASIC_INDEX_NAME)) {
+            prepareKnnIndex(
+                TEST_BASIC_INDEX_NAME,
+                Collections.singletonList(new KNNFieldConfig(TEST_KNN_VECTOR_FIELD_NAME_1, TEST_DIMENSION, TEST_SPACE_TYPE))
+            );
+            addKnnDoc(
+                TEST_BASIC_INDEX_NAME,
+                "1",
+                Collections.singletonList(TEST_KNN_VECTOR_FIELD_NAME_1),
+                Collections.singletonList(Floats.asList(testVector).toArray())
+            );
+            assertEquals(1, getDocCount(TEST_BASIC_INDEX_NAME));
+        }
+
+        if (TEST_MULTI_FIELD_INDEX_NAME.equals(indexName) && !indexExists(TEST_MULTI_FIELD_INDEX_NAME)) {
+            prepareKnnIndex(
+                TEST_MULTI_FIELD_INDEX_NAME,
+                List.of(
+                    new KNNFieldConfig(TEST_KNN_VECTOR_FIELD_NAME_1, TEST_DIMENSION, TEST_SPACE_TYPE),
+                    new KNNFieldConfig(TEST_KNN_VECTOR_FIELD_NAME_2, TEST_DIMENSION, TEST_SPACE_TYPE)
+                )
+            );
+            addKnnDoc(
+                TEST_MULTI_FIELD_INDEX_NAME,
+                "1",
+                List.of(TEST_KNN_VECTOR_FIELD_NAME_1, TEST_KNN_VECTOR_FIELD_NAME_2),
+                List.of(Floats.asList(testVector).toArray(), Floats.asList(testVector).toArray())
+            );
+            assertEquals(1, getDocCount(TEST_MULTI_FIELD_INDEX_NAME));
+        }
+
+        if (TEST_NESTED_INDEX_NAME.equals(indexName) && !indexExists(TEST_NESTED_INDEX_NAME)) {
+            prepareKnnIndex(
+                TEST_NESTED_INDEX_NAME,
+                Collections.singletonList(new KNNFieldConfig(TEST_KNN_VECTOR_FIELD_NAME_NESTED, TEST_DIMENSION, TEST_SPACE_TYPE))
+            );
+            addKnnDoc(
+                TEST_NESTED_INDEX_NAME,
+                "1",
+                Collections.singletonList(TEST_KNN_VECTOR_FIELD_NAME_NESTED),
+                Collections.singletonList(Floats.asList(testVector).toArray())
+            );
+            assertEquals(1, getDocCount(TEST_NESTED_INDEX_NAME));
+        }
+    }
+
+    @SneakyThrows
+    public void testBasicQuery() {
```

Comment:
suggestion: as the knowledge of tests will fade as the time passes, lets add the query and response objects in the java doc so that we know what to expect for all the tests present in this class.

Reply:
Will update

Reply:
Each response will be similar. I am going to just provide query.